### ISSUE TITLE
Shorter syntax for implicits

### ISSUE THREAD

**Issue by yallop** [ocamllabs/ocaml-modular-implicits#13]
Experiment with shorter syntax for implicits.  One possibility is to use Agda-style braces, so we'd have

``` ocaml
fun {M:X} -> ...      (* abstractions *)
f {M}                 (* applications *)
f {}                  (* application, leaving the actual instantiation to the search *)
{M:X} -> t            (* types *)
```

**Comment by bluddy** [comment]
I like it!

**Comment by yallop** [comment]
Fixed in  bd182d9ba61c2.
